server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -135,10 +137,14 @@ private SortedMap<String,String> makeRelative(Collection<String> candidates) {
 
   private void removeCandidatesInUse(GarbageCollectionEnvironment gce,
       SortedMap<String,String> candidateMap) {
+
+    Set<TableId> tableIdsBefore = gce.getCandidateTableIDs();
+    Set<TableId> tableIdsSeen = new HashSet<>();
     var refStream = gce.getReferences();
     Iterator<Reference> iter = refStream.iterator();
     while (iter.hasNext()) {
       Reference ref = iter.next();
+      tableIdsSeen.add(ref.getTableId());
```

Comment:
This `var` obscures the type and variable name `refStream` implies it's a Stream. But, then we call `.iterator()` on it. If it's actually a Stream, then there are better ways to manipulate it than doing `while (iter.hasNext()) { iter.next(); }`.

Reply:
Addressed in 1f8d37305789fa1540158e32b54da86a7af3ae87